Classes/Domain/Model/FileReference.php

```diff
@@ -38,10 +38,11 @@ class FileReference extends \TYPO3\CMS\Extbase\Domain\Model\FileReference {
 	protected $originalFileIdentifier;
 
 	/**
-	 * @param \TYPO3\CMS\Core\Resource\FileReference $originalResource
-	 */
-	public function setOriginalResource(\TYPO3\CMS\Core\Resource\FileReference $originalResource) {
-		$this->originalResource = $originalResource;
-		$this->originalFileIdentifier = (int)$originalResource->getOriginalFile()->getUid();
-	}
+	* @param \TYPO3\CMS\Core\Resource\ResourceInterface $originalResource
+	*/
+    	public function setOriginalResource(\TYPO3\CMS\Core\Resource\ResourceInterface $originalResource)
+    	{
+        	$this->originalResource = $originalResource;
+        	$this->originalFileIdentifier = (int)$originalResource->getOriginalFile()->getUid();
```

Comment:
ResourceInterface does not have the getOriginalFile() method.
Can you add a instance of FileReference check to avoid the fatal?